### DISCUSSION TITLE
#complete for struct literals with named fields

### DISCUSSION THREAD

**Discussion by JoshuaManton** [odin-lang/Odin#962] [Ideas/Requests]
Sometimes I like to do named fields just for clarity, especially for bigger structs. The following would help ensure that using named fields would be robust to new fields being added.

```
a := #complete Foo{x=1, y=4, z=9};
b := #complete Foo{x=1, y=4}; // compile error
```

Alternatively, the default could be #complete, like it is for non-named fields, then you would have to do `#complete Foo{x=1, y=4}`.

**Comment by jharler** [comment]
Is it worth making this part of the struct declaration instead of the variable declaration?

FWIW, I'd rather #complete not be the default.  Because the zero value is meant to be useful in Odin, in most cases #complete won't be necessary.

**Comment by ThisDevDane** [comment]
I agree with jharler that making `#complete` the default seems counter-productive to the "make the zero value" useful approach.
I also feel just having `#complete` even at struct init time seems to conflict with it. But if we look past that I do like this idea.
